csvcubed/tests/behaviour/steps/inspectcli.py

```diff
@@ -0,0 +1,117 @@
+    )
+    path = Path(context.csvw_metadata_json_path)
+    assert path.exists() == True
+
+
+@When("the Metadata File json-ld is loaded to a rdf graph")
+def step_impl(context):
+    metadata_processor = MetadataProcessor(context.csvw_metadata_json_path)
+    context.csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
+    assert context.csvw_metadata_rdf_graph is not None
+
+
+@When("the Metadata File is validated")
+def step_impl(context):
+    csvw_metadata_rdf_validator = MetadataValidator(context.csvw_metadata_rdf_graph)
+    (
+        context.valid_csvw_metadata,
+        context.csvw_type,
+    ) = csvw_metadata_rdf_validator.validate_and_detect_type()
+
+    assert context.valid_csvw_metadata == True
+
+
+@When("the Printables for data cube are generated")
+def step_impl(context):
+    metadata_printer = MetadataPrinter(
+        context.csvw_type,
+        context.csvw_metadata_rdf_graph,
+        context.csvw_metadata_json_path,
+    )
+    context.type_printable = metadata_printer.gen_type_info_printable()
+    context.catalog_metadata_printable = (
+        metadata_printer.gen_catalog_metadata_printable()
+    )
+    context.dsd_info_printable = metadata_printer.gen_dsd_info_printable()
+    context.codelist_info_printable = metadata_printer.gen_codelist_info_printable()
+
+    assert (
+        context.type_printable
+        and context.catalog_metadata_printable
+        and context.dsd_info_printable
+        and context.codelist_info_printable
+    )
+
+
+@When("the Printables for code list are generated")
+def step_impl(context):
+    metadata_printer = MetadataPrinter(
+        context.csvw_type,
+        context.csvw_metadata_rdf_graph,
+        context.csvw_metadata_json_path,
+    )
+    context.type_printable = metadata_printer.gen_type_info_printable()
+    context.catalog_metadata_printable = (
+        metadata_printer.gen_catalog_metadata_printable()
+    )
+
+    assert context.type_printable and context.catalog_metadata_printable
+
+
+@Then('the Type Printable should be "{type_printable}"')
+def step_impl(context, type_printable: str):
+    assert type_printable == context.type_printable
+
+
+@Then("the Catalog Metadata Printable should be")
+def step_impl(context):
+    assert _unformat_multiline_string(
+        context.catalog_metadata_printable
+    ) == _unformat_multiline_string(context.text)
+
+
+@Then("the Data Structure Definition Printable should be")
+def step_impl(context):
+    assert _unformat_multiline_string(
+        context.dsd_info_printable
+    ) == _unformat_multiline_string(context.text)
+
+
+@Then("the Code List Printable should be")
+def step_impl(context):
+    assert _unformat_multiline_string(
+        context.codelist_info_printable
+    ) == _unformat_multiline_string(context.text)
+
+
+@Given('a none existing test-case file "{csvw_metadata_file}"')
+def step_impl(_,csvw_metadata_file:str):
+    pass
```

Comment:
Do we need this?

Reply:
I have made an update here - Given can be used to store the file path so that the next step dont need the path as an arg.